c++/src/capnp/reconnect.h

```diff
@@ -28,35 +28,55 @@ CAPNP_BEGIN_HEADER
 
 namespace capnp {
 
+class AutoReconnectController {
+public:
+  // Reset the reconnect client to its initial state.
+  virtual void reset() = 0;
+};
+
+template <typename T>
+requires std::is_convertible_v<T, Capability::Client>
+struct AutoReconnectClient {
+  T cap;
+  kj::Rc<AutoReconnectController> controller;
```

Comment:
I don't think you can actually use `kj::Rc` on something that doesn't (publicly) inherit `Refcounted`? Presumably `.addRef()` wouldn't work at least.

```suggestion
  kj::Own<AutoReconnectController> controller;
```

Reply:
Yeah that's why I wasn't sure exactly what you were asking. You can still use the `kj::Rc` but, as you say, `addRef()` (for example) doesn't work.